### ISSUE TITLE
v6.db.transport.rest API status page inconsistent with my test results

### ISSUE THREAD

**Issue by LilaHexe0** [public-transport/transport.rest#33]
The API status page (https://stats.uptimerobot.com/57wNLs39M/793274556) claims 100% uptime on days where my personal monitoring (datadog) indicates otherwise:

![Screenshot 2024-12-17 at 15-09-45 Test on v6 db transport res  Datadog](https://github.com/user-attachments/assets/21dbcb37-5945-400a-838f-523c3d1a26b5)

Most/all of the failures are caused by 503 errors.

How exactly does the UptimeRobot check if the service is operational?

**Comment by traines-source** [comment]
I think the status page only indirectly, if at all, monitors whether HAFAS requests themselves are working. Maybe one could switch to monitoring the `/health` endpoint directly, of course entailing many more additional requests towards HAFAS.

The 503s you've been encountering are mostly due to errors on the HAFAS side, and it seems that the DB HAFAS mgate.exe endpoint will be shut off soon (see https://github.com/public-transport/hafas-client/issues/331 and 
https://github.com/schildbach/public-transport-enabler/issues/610)

**Comment by schaerfo** [comment]
I agree, using the `/health` endpoint for status monitoring would result in uptime stats that reflect real-world use cases of the API better.

**Comment by derhuerst** [comment]
At least with regards to DB's HAFAS API (and `v6.db.transport.rest`), this seems obsolete now that [it's likely shut-off for good](https://github.com/schildbach/public-transport-enabler/issues/610#issuecomment-2546009410).

However, let me make a more general point that applies to other HAFAS-based `*.transport.rest` APIs: **Obviously, the `/health` endpoint is not using caching. If you all use it to monitor availability of the API, you'll quickly exhaust the shared resource "requests from the server's single static IP", so you effectively prioritise your personal insight when the API is available over everyone's access to it.** To keep the rate of requests low, I don't see any solution to this other than making the `/health` endpoint private.

As an alternative, I suggest you to monitor "user-need-driven" requests (for actual public transport data), specifically e.g. their rate of success/error and the last successful one.

It might also be worthwhile to add [Prometheus](https://prometheus.io)-/[OpenMetrics](https://openmetrics.io)-compatible metrics to [`hafas-rest-api`](https://github.com/public-transport/hafas-rest-api) and expose them to the public, so you can ingest and monitor them.
